clippy_dev/src/issues_for_lint.rs

```diff
@@ -0,0 +1,154 @@
+            for (i, issue) in filter_issues(&issues, name, filter).enumerate() {
+                if i == 0 {
+                    println!("### `{}`\n", name);
+                }
+                println!("- [ ] #{} ({})", issue.number, issue.title)
+            }
+        },
+        Err(err) => eprintln!("{}", err),
+    }
+}
+
+pub fn run_all(filter: &[u32]) {
+    match open_issues() {
+        Ok(issues) => {
+            let mut lint_names = gather_all().map(|lint| lint.name).collect::<Vec<_>>();
+            lint_names.sort();
+            for name in lint_names {
+                let mut print_empty_line = false;
+                for (i, issue) in filter_issues(&issues, &name, filter).enumerate() {
+                    if i == 0 {
+                        println!("### `{}`\n", name);
+                        print_empty_line = true;
+                    }
+                    println!("- [ ] #{} ({})", issue.number, issue.title)
+                }
+                if print_empty_line {
+                    println!();
+                }
+            }
+        },
+        Err(err) => eprintln!("{}", err),
+    }
+}
+
+fn open_issues() -> Result<Vec<Issue>, Error> {
+    let github_token = env::var("GITHUB_TOKEN")?;
+
+    let mut headers = header::HeaderMap::new();
+    headers.insert(
+        header::AUTHORIZATION,
+        header::HeaderValue::from_str(&format!("token {}", github_token))?,
+    );
+    headers.insert(header::USER_AGENT, header::HeaderValue::from_static("ghost"));
+    let client = Client::builder().default_headers(headers).build()?;
+
+    let issues_base = "https://api.github.com/repos/rust-lang/rust-clippy/issues";
+
+    let mut issues = vec![];
+    let mut response = client
+        .get(issues_base)
+        .query(&[("per_page", "100"), ("state", "open"), ("direction", "asc")])
+        .send()?;
+    while let Some(link) = next_link(&response) {
+        issues.extend(
+            response
+                .json::<Vec<Issue>>()?
+                .into_iter()
+                .filter(|i| i.pull_request.is_none()),
+        );
+        response = client.get(&link).send()?;
+    }
+
+    Ok(issues)
+}
+
+fn filter_issues<'a>(issues: &'a [Issue], name: &str, filter: &'a [u32]) -> impl Iterator<Item = &'a Issue> {
+    let name = name.to_lowercase();
+    let separated_name = name.chars().map(|c| if c == '_' { ' ' } else { c }).collect::<String>();
+    let dash_separated_name = name.chars().map(|c| if c == '_' { '-' } else { c }).collect::<String>();
+
+    issues.iter().filter(move |i| {
+        let title = i.title.to_lowercase();
+        let body = i.body.to_lowercase();
+        !filter.contains(&i.number)
+            && (title.contains(&name)
+                || title.contains(&separated_name)
+                || title.contains(&dash_separated_name)
+                || body.contains(&name)
+                || body.contains(&separated_name)
+                || body.contains(&dash_separated_name))
+    })
```

Comment:
how about also checking the issue labels? If at some point in the future issues are then properly labeled with a `lint-*` could be easier to sort them

Reply:
Hm, we would then need over 300 labels for all of the lints and the issues had to be labeled correctly. Labeling issues by hand wouldn't be feasible. Maybe we could write an automation for such a labeling. But as long, as we don't have such a system in place, I don't think adding this to this tool makes much sense.

But The `lint-*`/`L-*` label idea is definitely interesting! Maybe we could add the labels of the issues in the output of this tool :thinking:

Reply:
> Maybe we could add the labels of the issues in the output of this tool

that's a great idea imo :+1:

As a positive side effect, people can easily start using the right label when reporting, decreasing in time the amount of "unlabeled" linting issues.

Reply:
Also, but I'm not sure it makes sense, as a starting point, I'd force feed the issueing ticketing system with all the labels with a quick `cURL`, so people can be instructed to use them immediately and we truncate immediately the input of "unlabeled" linting issues

Reply:
The thing is, that "normal" users can't label their issue by them self, but Collaborators/Members have to do this. There is also `@rustbot`, but we can't check this automatically. We'd could write a GitHub Actions Workflow, that scans a new issue for a lint name and labels it accordingly.

Reply:
ouch right I didn't think of that 🙃

Reply:
You can have issue templates that have a default label. I have some examples here: https://github.com/jyn514/rcc/tree/master/.github/ISSUE_TEMPLATE

Reply:
Unless I'm wrong, for issue templates to be useful we need to have the user be able to automatically apply one label out of dozens and that would require having dozens of templates, therefore a scary list of templates when the user creates the issue :-)

Reply:
Since we're soon completely migrated to GHA, we could implement the labeler workflow for issues and PRs, that would do that. But the question, if we want to have hundreds of labels still stands.

Reply:
If we can automate the label creation, I would not be against it personally

Reply:
I agree. So, I'd add a check for the issue label here (because imo it's a sensible default) and then we find a way to manage the issue creation workflow.